src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java

```diff
@@ -0,0 +1,286 @@
+    double lng = Double.parseDouble(getRequestParameterOrDefault(request, "lng", "0.0"));
+
+    // Coordinates 0 0 are located in the atlantic ocean, will be used to request worldwide cases
+    if (lat == 0.0 && lng == 0.0) {
+      LocationCases toReturn = new LocationCases("Worldwide", worldCases, dates);
+      Gson gson = new Gson();
+      String timeReportJson = gson.toJson(toReturn);
+      response.getWriter().println(timeReportJson);
+      return;
+    }
+
+    // Find closest report to coordinates in request
+    double minimumDistance = 1000.0;
+    LocLatLng potentialReport = null;
+    boolean usReport = false;
+    // First look through global reports
+    for (LocLatLng key : globalTimeReports.keySet()) {
+      double reportDistance = Math.abs(key.lat - lat) + Math.abs(key.lng - lng);
+      if (reportDistance < minimumDistance) {
+        minimumDistance = reportDistance;
+        potentialReport = key;
+      }
+    }
+    // Then look thorugh US reports
+    for (LocLatLng key : usTimeReports.keySet()) {
+      double reportDistance = Math.abs(key.lat - lat) + Math.abs(key.lng - lng);
+      if (reportDistance <= minimumDistance) {
+        usReport = true;
+        minimumDistance = reportDistance;
+        potentialReport = key;
+      }
+    }
+
+    // Return location name, cases, and dates
+    LocationCases toReturn;
+    if (usReport) {
+      toReturn =
+          new LocationCases(potentialReport.location, usTimeReports.get(potentialReport), dates);
+    } else {
+      toReturn = new LocationCases(
+          potentialReport.location, globalTimeReports.get(potentialReport), dates);
+    }
+    Gson gson = new Gson();
+    String timeReportJson = gson.toJson(toReturn);
+    response.getWriter().println(timeReportJson);
+  }
+
+  /**
+   * Establish connection to live Coivd-19 data set
+   */
+  private Scanner connectToData(String type) {
+    URL covidDataUrl = null;
+    HttpURLConnection connection = null;
+    int responseCode = 0;
+    Scanner scanner = null;
+
+    try {
+      String url =
+          "https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_"
+          + type + ".csv";
+      covidDataUrl = new URL(url);
+      connection = (HttpURLConnection) covidDataUrl.openConnection();
+      connection.setRequestMethod("GET");
+      responseCode = connection.getResponseCode();
+      if (responseCode == HttpURLConnection.HTTP_OK) {
+        scanner = new Scanner(covidDataUrl.openStream());
+      }
+    } catch (IOException e) {
+      System.out.println("Unable to open connection");
+    }
+    return scanner;
+  }
+
+  /**
+   * Fill up the hashmap with the name of the location and coordinates as the key
+   * and with an array consisting of the confirmed case numbers as the value
+   */
+  private void fillDataMap(Scanner scanner, HashMap<LocLatLng, ArrayList<Integer>> timeReports,
```

Comment:
same thing about the generic maps/list apply here. I think the idea is that you don't really care that it's a hashmap or an arraylist, the function would still work if it was a treemap and a linkedlist, but defining that you need them to be such implies that it's important (and in this case it's not, unless it is and I'm missing something)